src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts

```diff
@@ -163,207 +162,343 @@ const baseColumns = `
+    `;
+  } else if (isEventPathSQL) {
+    return 'and event.event_name not in (\'' + builtInEvents.filter(event => !eventNames.includes(event)).join('\',\'') + '\')';
+  } else {
+    return eventNameClause;
   }
+}
 
-  let conditionSql = '';
-  let columnSql = '';
-  for (const [index, eventNameAndSQLCondition] of eventNameAndSQLConditions.entries()) {
+export function _buildCommonPartSql(eventNames: string[], sqlParameters: SQLParameters,
+  isEventPathSQL: boolean = false, isNodePathAnalysis: boolean = false, isRetentionAnalysis: boolean = false) : string {
 
-    let normalConditionSql = eventNameAndSQLCondition.normalConditionSql;
-    if (normalConditionSql !== '') {
-      normalConditionSql = `and (${normalConditionSql}) `;
+  //If the query condition is simple use a different way to improve performance when fetch data
+  if (!_hasMoreThanOneNestConditionInEvent(sqlParameters)) {
+    return _buildCommonPartSqlSimple(eventNames, sqlParameters, isEventPathSQL, isNodePathAnalysis, isRetentionAnalysis);
+  } else {
+    const commonConditionSql = _getCommonConditionSql(sqlParameters, false, 'event.');
+    let allConditionSql = '';
+    if (!isRetentionAnalysis) {
+      allConditionSql = _getAllConditionSql(eventNames, sqlParameters, isEventPathSQL);
     }
+    const eventNameClause = _buildEventNameClause(eventNames, sqlParameters, isEventPathSQL, isNodePathAnalysis);
+
+    let eventJoinSql = '';
+    let userJoinSql = '';
+    const eventCondition = _hasEventCondition(sqlParameters);
+    let eventJoinTable = '';
+    let eventColList: string[] = [];
+    if (eventCondition.hasEventAttribute
+      || sqlParameters.pathAnalysis?.sessionType === ExplorePathSessionDef.SESSION
+    ) {
+      eventJoinSql = `join ${sqlParameters.schemaName}.clickstream_ods_events_parameter_view as event_param
+      on event.event_date = event_param.event_date and event.event_id = event_param.event_id
+      `;
 
-    let nestSqlPair0 = eventNameAndSQLCondition.nestSqlPair[0];
-
-    if (nestSqlPair0 !== '') {
-      nestSqlPair0 = `and (${nestSqlPair0}) `;
+      const eventAttributes = ['_session_id'];
+      eventAttributes.push(...eventCondition.eventAttributes);
+      const eventCommonColumnsSql = _buildCommonColumnsSql([...new Set(eventAttributes)], 'event_parameter_key', 'event_parameter_value');
+      eventColList = eventCommonColumnsSql.columns;
+      eventJoinTable = _buildEventJoinTable(sqlParameters.schemaName, eventJoinSql, eventCommonColumnsSql, commonConditionSql, eventNameClause);
     }
 
-    conditionSql = conditionSql.concat(`
-      ${index === 0? '' : 'or' } ( event_name = '${eventNameAndSQLCondition.eventName}' ${normalConditionSql} ${nestSqlPair0} )
-    `);
+    const userCondition = _hasUserCondition(sqlParameters);
+    let userJoinTable = '';
+    let userColList: string[] = [];
+    if (userCondition.hasUserAttribute) {
+      userJoinSql = `join ${sqlParameters.schemaName}.clickstream_user_attr_view as user_param
```

Comment:
We have `user-m-view` built by new transformer. Should you use the new view or update the user_attr view to use it?

Reply:
all sql is build on origin table and MVs in this PR.

Reply:
hold on this PR， will do more test on new tables.